openpype/modules/ftrack/plugins/publish/collect_ftrack_family.py

```diff
@@ -34,68 +33,65 @@ def process(self, instance):
             self.log.warning("No profiles present for adding Ftrack family")
             return
 
-        add_ftrack_family = False
-        task_name = instance.data.get("task",
-                                      legacy_io.Session["AVALON_TASK"])
-        host_name = legacy_io.Session["AVALON_APP"]
+        host_name = instance.context.data["hostName"]
         family = instance.data["family"]
+        task_name = instance.data.get("task")
 
         filtering_criteria = {
             "hosts": host_name,
             "families": family,
             "tasks": task_name
         }
-        profile = filter_profiles(self.profiles, filtering_criteria,
-                                  logger=self.log)
+        profile = filter_profiles(
+            self.profiles,
+            filtering_criteria,
+            logger=self.log
+        )
+
+        add_ftrack_family = False
+        families = instance.data.setdefault("families", [])
 
         if profile:
-            families = instance.data.get("families")
             add_ftrack_family = profile["add_ftrack_family"]
-
             additional_filters = profile.get("advanced_filtering")
             if additional_filters:
-                self.log.info("'{}' families used for additional filtering".
-                              format(families))
+                families_s = set(families) | {family}
```

Comment:
I think we could actually just replace the old `families` variable we had directly with this instead of moving it and renaming it. It being a `set` also helps the `if "ftrack" not in families` below too because O1 lookup in sets?

Reply:
Variable `families` is used to add `"ftrack"` family so we should not modify it, and we need to modify it to make sure `instance.data["family"]` is inside the family sequence. Thus new variable...

Reply:
But you are not using the mutable `families` variable to add `ftrack`. So current code doesn't really need it? Unless we expect `family` itself to be `ftrack` and not be in `families`?

Reply:
I see, I forgot to change it to `families.append("ftrack")`.

Reply:
it is a little thing, but could we rename `families_s` to `families_set`, it took me while to realize what it meant and it is only two letters long and saves time and mental power once is easier to read.